src/distPlus/distribution/Continuous.re

```diff
@@ -3,30 +3,45 @@ open Distributions;
 type t = DistTypes.continuousShape;
 let getShape = (t: t) => t.xyShape;
 let interpolation = (t: t) => t.interpolation;
-let make = (interpolation, xyShape, knownIntegralSum): t => {
+let make = (interpolation, xyShape, integralSumCache, integralCache): t => {
   xyShape,
   interpolation,
-  knownIntegralSum,
+  integralSumCache,
+  integralCache,
 };
-let shapeMap = (fn, {xyShape, interpolation, knownIntegralSum}: t): t => {
+let shapeMap = (fn, {xyShape, interpolation, integralSumCache, integralCache}: t): t => {
```

Comment:
Do we really need to store the integralSumCache? I would have expected this to be the same as the last value of the integralCache, meaning we could calculate it on demand.

Reply:
Also, just wanted to note; having the integral be here seems pretty fair to me, though it is a significant change.

Reply:
integralSumCache is really cheap to keep track of – we know it starts out as 1.0 for the symbolic leaf nodes, and then gets scaled up and down by multimodals. So we can often normalize using integralSumCache without having to ever integrate at all. That was the initial motivation for it, and I think it still holds.

Reply:
So to be clear, is it only useful in specific situations where you can't save the integral, but have calculated the integralSum? If that's true, I could see storing it to make that obvious, like,
```
type cache = [| `None | `integralSum(float) | `integral(xyShape) |]
```